### DISCUSSION TITLE
Add "mixed" as a valid value for the "fraction" option for math.format

### DISCUSSION THREAD

**Discussion by jermowery** [josdejong/mathjs#3053] [Ideas]
Right now `math.format` allows for an `options` object to be passed as a second parameter.
This object may contain a `fraction` property with the values: "decimal" or "ratio"
[Docs ](https://mathjs.org/docs/reference/functions/format.html#where:~:text=fraction%3A%20string.%20Available%20values%3A%20%E2%80%98ratio%E2%80%99%20(default),output%200.(3)%20when%20%E2%80%98decimal%E2%80%99%20is%20configured.)
[Code](https://github.com/josdejong/mathjs/blob/e1dce7bf4f81c95db86f1e4094d61ce86b275323/src/utils/string.js#L62)

This FR is to add a new value for this property "mixed"

When "mixed" is specified then numbers will be formatted as mixed numbers, eg:

```
math.format(math.parse("1.25"), {fraction: "mixed"}) // 1 1/4
math.format(math.parse("1.25"),  {fraction: "decimal"}) // 1.25
math.format(math.parse("1.25"), {fraction: "ratio"}) // 5/4
math.format(math.parse("1.25")) // 1.25
```

Implementation should be as simple as adding another branch in the code linked above

**Comment by josdejong** [comment]
Interesting idea Jeremy, thanks for sharing. I like it.

A few thoughts:

1. I think the default behavior should stay the same, i.e. outputting decimals when option `fraction` is not configured. So `math.format(math.parse("1.25"))` should output `1.25` and not `5/4` as in your example, right?
2. I think so far all outputs of `math.format` can be fed directly into the expression parser again and evaluated. This option "mixed" would break that, since `1 1/4` is no valid input for the parser. That may be acceptable, and we could also look into making this valid syntax (maybe tricky with implicit multiplication, we'll have to see).

**Comment by jermowery** [comment]
1. Sorry, I thought ratio was the default. My intention in the example above was to keep the default the same. I will update the examples

Thanks for the reply!

**Comment by gwhitney** [comment]
Given the potential difficulties/API decisions/implications entailed in adding such an option which might produce output that can't be read back in, I think this belongs in a Discussion.
